### PR TITLE
Fix the support for enum types in the ResultSetMappingBuilder

### LINKED ISSUE #11017: ResultSetMappingBuilder does not properly support enumType
### Bug Report

|    Q        |   A
|------------ | ------
| BC Break    | no
| Version     | 2.16.2

#### Summary

`Doctrine\ORM\Query\ResultSetMappingBuilder` is not aware of the `enumType` feature

#### Current behavior

When using `\Doctrine\ORM\Query\ResultSetMappingBuilder::addRootEntityFromClassMetadata` or `\Doctrine\ORM\Query\ResultSetMappingBuilder::addJoinedEntityFromClassMetadata` with an entity that uses `enumType` for some fields, the info about a field needing the enum conversion is not added automatically in the ResultSetMapping based on the metadata.

#### Expected behavior

The enum conversion should be added automatically in the ResultSetMapping based on the metadata, like other features defined in the metadata.

### PR DESCRIPTION
**Pull request by stof**
Fixes https://github.com/doctrine/orm/issues/11017

### CODE REVIEW

**Comment by greg0ire** [comment]
The failures look unrelated, but I think they actually are somehow: PHPUnit is happy with `origin/2.16.x` right now.

File: lib/Doctrine/ORM/Query/ResultSetMappingBuilder.php [hunk 1]
```diff
@@ -154,6 +154,11 @@ protected function addAllClassFields($class, $alias, $columnAliasMap = [])
             }
 
             $this->addFieldResult($alias, $columnAlias, $propertyName);
+
+            $enumType = $classMetadata->getFieldMapping($columnName)['enumType'] ?? null;
```
Comment (greg0ire) [inline lib/Doctrine/ORM/Query/ResultSetMappingBuilder.php]: This is the line that is causing the issue. Getting the field mapping seems to have unwanted consequences.

Comment (greg0ire) [reply lib/Doctrine/ORM/Query/ResultSetMappingBuilder.php]: Hang on… you're calling a field mapping with a column name here. This is why these tests are failing.

Comment (greg0ire) [reply lib/Doctrine/ORM/Query/ResultSetMappingBuilder.php]:
```suggestion
            $enumType = $classMetadata->getFieldMapping($propertyName)['enumType'] ?? null;
```

This leaves one failure (so no errors), with the test you're introducing.

File: tests/Doctrine/Tests/ORM/Functional/Ticket/GH11017/GH11017Test.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+        $tableName = $this->_em->getClassMetadata(GH11017Entity::class)->getTableName();
+        $sql       = sprintf('SELECT %s FROM %s e WHERE id = :id', $rsm->generateSelectClause(), $tableName);
+
+        $query = $this->_em->createNativeQuery($sql, $rsm)
+            ->setParameter('id', $entity1->id);
+
+        $entity1Reloaded = $query->getSingleResult(AbstractQuery::HYDRATE_ARRAY);
+        self::assertNotNull($entity1Reloaded);
+        self::assertSame($entity1->field, $entity1Reloaded['field']);
```
Comment (greg0ire) [inline tests/Doctrine/Tests/ORM/Functional/Ticket/GH11017/GH11017Test.php:47]:
```suggestion
        self::assertSame($entity1->field->value, $entity1Reloaded['field']);
```

File: lib/Doctrine/ORM/Query/ResultSetMappingBuilder.php (hunk 1 above)
Comment (stof) [reply lib/Doctrine/ORM/Query/ResultSetMappingBuilder.php]: indeed, I messed up (and my test was running fine because it was a case where both were equal)

**Comment by stof** [comment]
@greg0ire this is fixed now

**Review by greg0ire — APPROVED** [review]

**Comment by greg0ire** [comment]
Thanks @stof !

**Pull request merged**